fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinReorderDP.java

```diff
@@ -107,12 +97,32 @@ private GroupInfo getBestExpr(Set<OptExpression> joinKeys, List<GroupInfo> group
         return bestPlan;
     }
 
-    private Set<Set<Integer>> generatePartitions(int totalNodes) {
-        checkArgument(totalNodes > 1, "totalNodes must be greater than 1");
-        Set<Integer> numbers = IntStream.range(0, totalNodes).boxed().collect(toImmutableSet());
-        return powerSet(numbers).stream()
+    private List<BitSet> generatePartitions(BitSet totalNodes) {
+        Set<Integer> numbers = IntStream.range(0, totalNodes.cardinality()).boxed().collect(toImmutableSet());
+        Set<Set<Integer>> sets = powerSet(numbers).stream()
```

Comment:
why not use bitset to generate powerSet

Reply:
Because the input parameter of powerSet is Set<Integer>, BitSet cannot be used, so a conversion is here